### ISSUE TITLE
IDISP003 should not warn hwen assigning out parameter in if returns

### ISSUE THREAD

**Issue by JohanLarsson** [DotNetAnalyzers/IDisposableAnalyzers#124]
```cs
namespace RoslynSandbox
{
    using System.IO;

    class Foo
    {
        private static bool TryGetStream(string fileName, out Stream result)
        {
            if (File.Exists(fileName))
            {
                result = File.OpenRead(fileName);
                return true;
            }

            if (File.Exists(fileName))
            {
                result = File.OpenRead(fileName);
                return true;
            }

            result = null;
            return false;
        }
    }
}
```
